jumanji/environments/packing/flat_pack/reward.py

```diff
@@ -0,0 +1,153 @@
+        self,
+        state: State,
+        action: chex.Numeric,
+        next_state: State,
+        is_valid: bool,
+        is_done: bool,
+    ) -> chex.Numeric:
+        """Compute the reward based on the current state, the chosen action,
+        whether the action is valid and whether the episode is terminated.
+        """
+
+
+class CellDenseReward(RewardFn):
+    """Reward function for the dense reward setting.
+
+    This reward returns the number of non-zero cells in a placed block normalised
+        by the total number of cells in the grid. This means that the maximum possible
+        episode return is 1. That is to say that, in the case of this reward, an agent
+        will optimise for maximal area coverage in the the grid.
+    """
+
+    def __call__(
+        self,
+        state: State,
+        action: chex.Numeric,
+        next_state: State,
+        is_valid: bool,
+        is_done: bool,
+    ) -> chex.Numeric:
+        """Compute the reward based on the current state, the chosen action,
+        whether the action is valid and whether the episode is terminated.
+
+        Note here, that the action taken is not the raw action received from the
+        agent, but the block the agent opted to place on the grid.
+        """
+
+        del is_done
+        del next_state
+        del state
+
+        action_ones = action != 0.0
+        num_rows, num_cols = action_ones.shape
+
+        reward = jax.lax.cond(
+            is_valid,
+            lambda: jnp.sum(action_ones, dtype=jnp.float32) / (num_rows * num_cols),
+            lambda: jnp.float32(0.0),
+        )
+
+        return reward
+
+
+class BlockDenseReward(RewardFn):
+    """Reward function for the dense reward setting.
+
+    This reward will give a normalised reward for each block placed on the grid
+    with each block being equally weighted. This implies that each placed block
+    will have a reward of `1 / num_blocks` and the maximum possible episode return
+    is 1. That is to say that, in the case of this reward, an agent will optimise
+    for placing as many blocks as possible on the grid.
+    """
+
+    def __call__(
+        self,
+        state: State,
+        action: chex.Numeric,
+        next_state: State,
+        is_valid: bool,
+        is_done: bool,
+    ) -> chex.Numeric:
+        """Compute the reward based on the current state, the chosen action,
+        whether the action is valid and whether the episode is terminated.
+
+        Note here, that the action taken is not the raw action received from the
+        agent, but the block the agent opted to place on the grid.
+        """
+
+        del is_done
+        del next_state
+        del action
+
+        num_blocks = state.num_blocks
+        del state
+
+        reward = jax.lax.cond(
+            is_valid,
+            lambda: jnp.float32(1.0 / num_blocks),
+            lambda: jnp.float32(0.0),
+        )
+
+        return reward
+
+
+class SparseReward(RewardFn):
```

Comment:
Very nice suggestion! I do agree that the current one is very very sparse. But I do not feel like it is critical to implement another reward at the moment since our training curves are on the dense reward. Keen to hear your thoughts?

Reply:
That is a fair point. I think we can just remove it in that case.

Reply:
Removed it :)